src/data/zmod/basic.lean

```diff
@@ -81,6 +81,17 @@ instance (n : ℕ) : char_p (zmod n) n :=
     rw [val_nat_cast, val_zero, nat.dvd_iff_mod_eq_zero],
   end }
 
+@[simp] lemma add_order_of_one (n : ℕ) : add_order_of (1 : zmod n) = n :=
+char_p.eq _ (char_p.eq_add_order_of_one _) (zmod.char_p n)
+
+@[simp] lemma add_order_of_coe (a : ℕ) {n : ℕ} (n0 : n ≠ 0) :
```

Comment:
According to your above lemma, this is still true for `n = 0`.
```suggestion
@[simp] lemma add_order_of_coe (a : ℕ) (n : ℕ) :
```

Reply:
I'm not sure about this:
```lean
example : add_order_of (0 : zmod 0) ≠ 0 / nat.gcd 0 0 :=
by simp
```
I could probably include more comprehensive assumptions to cover the case where either `a` or `n` is non-zero, or add a separate lemma with `a ≠ 0` as an assumption and no assumption on `n`.  Is one of these options a good idea?

Reply:
I think that the culprit is `0 / 0 = 0`, instead of `1`.

Reply:
@YaelDillies, I added another lemma to cover the case `a ≠ 0`.  I think that it is better to have two lemmas, covering all cases, than a single lemma with an `\or` assumption.  If you have a different opinion, feel free to say so!

Reply:
I'm very happy to have two lemmas, but this is very confusing to me because the above lemma makes me think that `add_order_of (0 : zmod 0) = add_order_of (1 : zmod 0) = 0 = 0 / gcd 0 0`.

Reply:
Isn't the confusion that `zmod 0` "is" `nat`, though?  So, `(0 : zmod 0)` has order 1, since `1 * 0 = 0`, while `(1 : zmod 0)` has order `0`, since infinite order is encoded as `0`.

Am I also confused or does this clear your confusion?  :upside_down_face:

Reply:
Alternatively, when `zmod ??` has zero elements, that is, when `?? = 1`!), then your argument is correct and the proof above shows it!

Reply:
Of course 🤦